src/components/ColorPicker/ColorPicker.vue

```diff
@@ -0,0 +1,190 @@
+ -
+ - This program is distributed in the hope that it will be useful,
+ - but WITHOUT ANY WARRANTY; without even the implied warranty of
+ - MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ - GNU Affero General Public License for more details.
+ -
+ - You should have received a copy of the GNU Affero General Public License
+ - along with this program. If not, see <http://www.gnu.org/licenses/>.
+ -
+ -->
+
+ <docs>
+</docs>
+
+<template>
+	<div class="color-picker">
+		<div v-if="!advanced" class="color-picker-simple">
+			<div v-for="(color, index) in palette" :key="index"
+				:style="{'background-color': color}"
+				class="color-picker-simple-color-circle" />
+		</div>
+		<transition name="slide">
+			<Chrome v-if="advanced" v-model="advColor" class="color-picker-advanced"
+				:disable-alpha="true"
+				:disable-fields="true" />
+		</transition>
+		<div class="color-picker-navigation">
+			<button v-if="advanced" class="color-picker-navigation-button back"
+				@click='handleBack' />
+			<button v-if="advanced" class="color-picker-navigation-button confirm"
+				@click='handleConfirm' />
+			<button v-if="!advanced" class="color-picker-navigation-button more-settings"
+				@click='handleMoreSettings' />
+		</div>
+	</div>
+</template>
+
+<script>
+import { Chrome } from 'vue-color'
+
+export default {
+	name: 'ColorPicker',
+	components: {
+		Chrome
+	},
+
+	data() {
+		return {
+			advColor: '#194d33',
+			advanced: false,
+			palette: ['#CFCFCF', '#CFCFCF', '#CFCFCF', '#CFCFCF', '#CFCFCF', '#CFCFCF', '#CFCFCF', '#CFCFCF', '#CFCFCF', '#CFCFCF', '#CFCFCF', '#CFCFCF', '#CFCFCF', '#CFCFCF', '#CFCFCF', '#CFCFCF']
+		}
+	},
+	methods: {
+		handleConfirm() {
+			alert(`You've chosen the color ${this.color.hex}`)
+		},
+		handleBack() {
+			this.advanced = false
+		},
+		handleMoreSettings() {
+			this.advanced = true
+		}
+	}
+}
+
+</script>
+
+<style lang="scss" scoped>
+@import '~Fonts/scss/iconfont-vue';
+
+.color-picker {
+	width: 164px;
+	height: 208px;
+	margin: 100px;
+	box-shadow: 0 0 2px rgba(0,0,0,.3), 0 4px 8px rgba(0,0,0,.3);
+	padding: 16px;
+	border-radius: 3px;
+	display: flex;
+	flex-direction: column;
+	justify-content: space-between;
+	align-content: flex-end;
+	overflow: hidden;
+	box-sizing: content-box !important;
+	&-simple {
+		display: grid;
+		grid-template-columns: repeat(4, 41px);
+		grid-template-rows: repeat(4, 41px);
+		&-color-circle {
+			width: 27px;
+			height: 27px;
+			border-radius: 13px;
+			margin:auto;
+		}
+	}
+	&-advanced {
+		box-shadow: none !important;
+	}
+	&-navigation {
+		display: flex;
+		flex-direction: row;
+		justify-content: space-between;
+		&-button {
+			justify-self: flex-end;
+			&.back{
+				@include iconfont('arrow-left');
+			}
+			&.confirm {
+				@include iconfont('checkmark');
+			}
+			&.more-settings {
+				@include iconfont('more');
+				margin-left: auto;
+			}
+		}
+	}
+}
+
+::v-deep .vc-chrome {
+	width: 164px;
```

Comment:
if the content is properly designed and have proper sizing and layout, youshould not need to have custom width. width: auto is enough.

Reply:
Same line 84 ;)

Reply:
These are needed to overrule the defaults coming from the vue-color library, I'm afraid I cannot take them out